module/documents/actor/hit-dice.mjs

```diff
@@ -10,12 +10,8 @@ export default class HitDice {
     this.actor = actor;
 
     for ( const item of Object.values(actor.classes) ) {
-      if ( /^d\d+$/.test(item.system.hitDice) ) {
-        this.classes.add(item);
-        this.value += item.system.levels - item.system.hitDiceUsed;
-        this.max += item.system.levels;
-        this.sizes.add(parseInt(item.system.hitDice.slice(1)));
-      }
+      this.classes.add(item);
```

Comment:
The data validation was performed in the class item already, so it's not necessary here. The `value` and `max` properties are moved to getters instead due to data preparation order (`additional` is not evaluated yet here).